hs_core/hydroshare/resource.py

```diff
@@ -722,16 +725,32 @@ def delete_resource_file(pk, filename_or_id, user):
 
     return filename_or_id
 
+def get_resource_doi(res_id, pending):
+    doi_str = "http://dx.doi.org/10.4211/hs.{shortkey}".format(shortkey=res_id)
+    if pending:
+        return "{doi}pending".format(doi=doi_str)
+    else:
+        return doi_str
+
+
+def get_activated_doi(doi):
+    idx = doi.find('pending')
+    if idx >= 0:
+        return doi[:idx]
+    else:
+        return doi
 
-def publish_resource(pk):
+def publish_resource(user, pk):
```

Comment:
I agree with @dtarb - we need to handle a failure on our side.  The user should not have to do another publication request.  

We could re-queue the issue for a period of time (like 3 hours) and if it's still unsuccessful you could fire an email to the user saying that their request has been queued, but that the DOI creation service is currently experiencing issues.  One would hope that if we are paying Crossref for a service that their production system is reliable and this won't happen.  But, if it does, we should just notify the user.  We may have to have those resources remain in a queue until Crossref becomes responsive again, and it might be good to have an admin function to restart the registration process for any DOIs that get stuck.